### ISSUE TITLE
Use maven for dependency management

### ISSUE THREAD

**Issue by imagejan** [biocompibens/SME#2]
I suggest to [mavenize](https://imagej.net/Maven) the java plugin for dependency management, and extend [`pom-scijava`](https://github.com/scijava/pom-scijava/) as a parent.

This will allow to drop the `org/apache/commons/math3` folder from this repository, and avoid shipping an [uber-jar](https://imagej.net/Uber-JAR) which might potentially lead to issues with version skew for users in the future. (Fiji/ImageJ currently ships [`commons-math3-3.6.1.jar`](https://github.com/scijava/pom-scijava/blob/master/pom.xml#L583-L584) already)

**Comment by erexhepa** [comment]
yes absolutely maven is the way to go and extention to pom-scijava it is needed. 

That is true that the library is shipped with Fiji but it is not for ImageJ. The only issue we had with the apache library is that you rely on them for backward compatibility and if at some point this is not the case then you need a version of the commons-math library to be able to build the plugin. 

Even with Fiji the update of the apache math library (from what I understood) ensure Fiji compatibility but not that of the plugins that are not part of the base distribution.
